Complete Release notes (#1151)
Avaiga/taipy-doc: docs/release-notes/index.md

<h4><strong><code>taipy</code></strong> 4.0.0</h4>

> - The User Experience of the Scenario and Data management Visual elements have been greatly

Reviewer: Thanks for that!

- The `Scenario.export()` and `taipy.export_scenario()` functions have been transferred from the
  Community edition to the Enterprise edition as it is more suitable for enterprise applications.
- The production mode and the migration configuration of the version management system has been
  transferred from the Community edition to the Enterprise edition as it is more suitable for
  enterprise applications.
- Support for the SQL repository was removed. Taipy Community edition now only supports the
  `filesystem` repository type.<br/>
  See [issue #1513](https://github.com/Avaiga/taipy/issues/1513).
- Support for different encodings in `S3ObjectDataNode^`.<br/>
  See [issue #680](https://github.com/Avaiga/taipy/issues/680).
- Reading an `ExcelDataNode^` is more consistent across the various expose types.<br/>
  See [issue #796](https://github.com/Avaiga/taipy/issues/796).
- Two scenarios belonging to the same cycle can now have the same tag.<br/>
  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).<br/>
- The custom properties of a `Scenario` are not exposed as attribute anymore.<br/>
  See [issue #1572](https://github.com/Avaiga/taipy/issues/1572).

> - Methods and functions returning booleans and related to entities now return

Reviewer: ```suggestion
- Methods and functions returning a Boolean value and related to entities now return
```

>   `ReasonCollection^` as a set of `Reason^` instances. Each reason contains a message

Reviewer: ```suggestion
  a `ReasonCollection^` object, which is a set of `Reason^` instances. Each reason contains a message
```

>   explaining why the boolean is `False`. The collection is empty if the boolean is `True`

Reviewer: ```suggestion
  explaining why the boolean is `False`. The collection is empty if the Boolean is `True`
```

>   explaining why the boolean is `False`. The collection is empty if the boolean is `True`

Reviewer: ```suggestion
  explaining why the return value is False. The collection is empty if the returned value is True
```

>   explaining why the boolean is `False`. The collection is empty if the Boolean is `True`

Author: ```suggestion
  explaining why the returned value is `False`. The collection is empty if the value is `True`
```

  .<br/>

>   Examples: `is_deletable^`, `exists^`, `is_readable^` etc.

Reviewer: ```suggestion
  Examples: `is_deletable()^`, `exists()^`, `is_readable()^` etc.
```

  See [issue #1568](https://github.com/Avaiga/taipy/issues/1568).
- The `Config.check()^` method now raises `ERROR` issues if any data node, task, or sequence of
  a `ScenarioConfig^` has the same configuration id as another one in the same `ScenarioConfig^`,
  or any additional property of any configuration has the same name as one of the attributes
  of the configuration class.<br/>
  See [issue #1696](https://github.com/Avaiga/taipy/issues/1696) and
  [issue #411](https://github.com/Avaiga/taipy/issues/411).<br/>
  For more information on checkers, please refer to
  [Configuration checker](../userman/advanced_features/configuration/config-checker.md).

<h4><strong><code>taipy-templates</code></strong> 4.0.0 </h4>

- The *--template* option of the `taipy create` command is now renamed to *--application* option

>   to correctly reflect the application template to use when creating a new Taipy application. <br/>

Reviewer: ```suggestion
  to correctly reflect the application template to use when creating a new Taipy application.<br/>
```

>   See[issue #1472](https://github.com/Avaiga/taipy/issues/1472)

Reviewer: ```suggestion
  See [issue #1472](https://github.com/Avaiga/taipy/issues/1472)
```

>   See[issue #1472](https://github.com/Avaiga/taipy/issues/1472)

Reviewer 2: Add a space between "See" and "["

Author: ```suggestion
  See [issue #1472](https://github.com/Avaiga/taipy/issues/1472)
```

- `DataNode.is_up_to_date()^` raises an error when the data node has never been written.<br/>
  See [issue #1198](https://github.com/Avaiga/taipy/issues/1198).

> - Reload the cache on all _build_manager() methods when the *repository_type* is changed.<br/>

Reviewer: ```suggestion
- Reload the cache on all *_build_manager()* methods when the *repository_type* is changed.<br/>
```

  See [issue #1692](https://github.com/Avaiga/taipy/pull/1692)

<h4><strong><code>taipy-templates</code></strong> 4.0.0</h4>

> - First cli option of the `taipy create` command is skipped if it's before the positional

Reviewer: ```suggestion
- The first cli option of the `taipy create` command is skipped if it's before the positional
```
